pulp_rpm/app/models/repository.py

```diff
@@ -321,26 +326,34 @@ class RpmDistribution(PublicationDistribution):
     def content_handler(self, path):
         """Serve config.repo and public.key."""
         if path == self.repository_config_file_name:
-            val = f"""[{self.name}]
-enabled=1
-baseurl={settings.CONTENT_ORIGIN}{settings.CONTENT_PATH_PREFIX}{self.base_path}/
-gpgcheck=0
-"""
+            base_url = f"{settings.CONTENT_ORIGIN}{settings.CONTENT_PATH_PREFIX}{self.base_path}/"
+            publication = self.publication.cast()
+            val = textwrap.dedent(
+                f"""\
+                [{self.name}]
+                enabled=1
+                baseurl={base_url}
+                gpgcheck={publication.gpgcheck}
+                """
+            )
             repository_pk = self.publication.repository.pk
             repository = RpmRepository.objects.get(pk=repository_pk)
             signing_service = repository.metadata_signing_service
             if signing_service is None:
-                val += 'repo_gpgcheck=0'
+                val += f"repo_gpgcheck={publication.repo_gpgcheck}\n"
             else:
                 gpgkey_path = urllib.parse.urljoin(
                     settings.CONTENT_ORIGIN, settings.CONTENT_PATH_PREFIX
                 )
                 gpgkey_path = urllib.parse.urljoin(gpgkey_path, self.base_path, True)
                 gpgkey_path += '/repodata/public.key'
 
-                val += f"""repo_gpgcheck=1
-gpgkey={gpgkey_path}
-"""
+                val += textwrap.dedent(
+                    f"""\
+                    repo_gpgcheck={publication.repo_gpgcheck}
```

Comment:
based on your logic repo_gpgcheck is written to the config anyway so i sugges to move it to line L331